django/db/backends/mysql/schema.py

```diff
@@ -69,6 +69,14 @@ def _field_should_be_indexed(self, model, field):
                 field.get_internal_type() == 'ForeignKey' and
                 field.db_constraint):
             return False
+        db_type = field.db_type(self.connection)
+        if (db_type is not None and
+                db_type.lower() in {
```

Comment:
Should this use `skip_default()`? Or perhaps the condition should be moved into some private method with a more semantic name and used here and in `skip_default()`.